java/ql/src/Security/CWE/CWE-113/NettyResponseSplitting.ql

```diff
@@ -13,8 +13,21 @@
 
 import java
 
-from ClassInstanceExpr new
-where
-  new.getConstructedType().hasQualifiedName("io.netty.handler.codec.http", "DefaultHttpHeaders") and
-  new.getArgument(0).getProperExpr().(BooleanLiteral).getBooleanValue() = false
-select new, "Response-splitting vulnerability due to verification being disabled."
+abstract private class InsecureNettyObjectCreation extends ClassInstanceExpr { }
+
+private class InsecureDefaultHttpHeadersClassInstantiation extends InsecureNettyObjectCreation {
+  InsecureDefaultHttpHeadersClassInstantiation() {
+    getConstructedType().hasQualifiedName("io.netty.handler.codec.http", "DefaultHttpHeaders") and
+    getArgument(0).getProperExpr().(BooleanLiteral).getBooleanValue() = false
```

Comment:
```suggestion
    getArgument(0).getProperExpr().(CompileTimeConstantExpr).getBooleanValue() = false
```

Would this still work or would this be wrong?
If this works, I should do the same thing below as well!

Reply:
That works, yes, and should be a strict improvement.  With that you can even simplify and drop the `.getProperExpr()` as that is included in `CompileTimeConstantExpr`.

Reply:
```suggestion
    getArgument(0).(CompileTimeConstantExpr).getBooleanValue() = false
```